### PR TITLE
[SYCL] Change pattern to reflect the change in sycl-ls

### PR DESCRIPTION
**Pull request by bso-intel**
In https://github.com/intel/llvm/pull/3611, I changed the device type to lower case which was necessary to show the SYCL_DEVICE_FILTER prefix for each device listing from sycl-ls.
It caused several tests failures that expected 'GPU' instead of 'gpu'.

Signed-off-by: Byoungro So <byoungro.so@intel.com>

### CODE REVIEW

**Review by smaslov-intel — COMMENTED** [review]
Why can't we make this, and other fields of SYCL_DEVICE_FILTER not case-sensitive?

**Comment by bso-intel** [comment]
> Why can't we make this, and other fields of SYCL_DEVICE_FILTER not case-sensitive?

Currently, the implementation of `std::ostream device_filter::operator<<()` prints only lower-cases.
Users can set SYCL_DEVICE_FILTER whatever cases they want, but our test only look for specific patterns from the `sycl-ls` print output.
So, to make these tests pass, I had to change the pattern to lower cases.

**Comment by smaslov-intel** [comment]
> > Why can't we make this, and other fields of SYCL_DEVICE_FILTER not case-sensitive?
> 
> Currently, the implementation of `std::ostream device_filter::operator<<()` prints only lower-cases.
> Users can set SYCL_DEVICE_FILTER whatever cases they want, but our test only look for specific patterns from the `sycl-ls` print output.
> So, to make these tests pass, I had to change the pattern to lower cases.

I mean I'd like to see a RT change to accept upper-case values instead of this change.

**Comment by bso-intel** [comment]
> > > Why can't we make this, and other fields of SYCL_DEVICE_FILTER not case-sensitive?
> > 
> > 
> > Currently, the implementation of `std::ostream device_filter::operator<<()` prints only lower-cases.
> > Users can set SYCL_DEVICE_FILTER whatever cases they want, but our test only look for specific patterns from the `sycl-ls` print output.
> > So, to make these tests pass, I had to change the pattern to lower cases.
> 
> I mean I'd like to see a RT change to accept upper-case values instead of this change.

Yes, RT accepts upper-case values (or any combination works).
But, when we print device filter, we only print the device type in lower case, which was used in sycl-ls.

Please note that this test is only trying to match the string pattern that 'sycl-ls' prints.
It is not testing what string SYCL_DEVICE_FILTER is accepting.

**Review by smaslov-intel — DISMISSED** [review]

**Review by vladimirlaz — DISMISSED** [review]

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp [hunk 1]
```diff
@@ -12,16 +12,16 @@
...
-// RUN: env SYCL_DEVICE_FILTER=cpu:0,acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-GPU-NOT,CHECK-ACC-NOT,CHECK-HOST
-// RUN: env CL_CONFIG_CPU_EMULATE_DEVICES=2 SYCL_DEVICE_FILTER=cpu:0,acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-GPU-NOT,CHECK-ACC,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=cuda:cpu,opencl:gpu:0,level_zero:acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU-NOT,CHECK-GPU-NOT,CHECK-ACC-NOT,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=cpu,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=acc,host %t.out | FileCheck %s --check-prefixes=CHECK-ACC,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=host %t.out | FileCheck %s --check-prefixes=CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=gpu,host %t.out | FileCheck %s --check-prefixes=CHECK-GPU,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=cpu,acc,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-ACC,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=cpu,acc,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-HOST
```
Comment (vladimirlaz) [inline SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: check for acc is missed

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp [hunk 2]
```diff
@@ -12,16 +12,16 @@
...
+// RUN: env CL_CONFIG_CPU_EMULATE_DEVICES=2 SYCL_DEVICE_FILTER=cpu,acc,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-ACC,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=cuda:cpu,opencl:gpu,level_zero:acc,host %t.out | FileCheck %s --check-prefixes=CHECK-GPU,CHECK-HOST
 //
 // CHECK-ACC: Device: acc
-// CHECK-CPU: Device: cpu
 // CHECK-GPU: Device: gpu
+// CHECK-CPU: Device: cpu
```
Comment (vladimirlaz) [inline SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: Why this line was moved? It does not look like intended change,

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp [hunk 3]
```diff
@@ -12,16 +12,16 @@
...
-// RUN: env SYCL_DEVICE_FILTER=acc,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU-NOT,CHECK-GPU-NOT,CHECK-ACC,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=host %t.out | FileCheck %s --check-prefixes=CHECK-CPU-NOT,CHECK-GPU-NOT,CHECK-ACC-NOT,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=gpu,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU-NOT,CHECK-GPU,CHECK-ACC-NOT,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=cpu,acc,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-GPU-NOT,CHECK-ACC,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=cpu:0,acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-GPU-NOT,CHECK-ACC-NOT,CHECK-HOST
-// RUN: env CL_CONFIG_CPU_EMULATE_DEVICES=2 SYCL_DEVICE_FILTER=cpu:0,acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-GPU-NOT,CHECK-ACC,CHECK-HOST
-// RUN: env SYCL_DEVICE_FILTER=cuda:cpu,opencl:gpu:0,level_zero:acc:1,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU-NOT,CHECK-GPU-NOT,CHECK-ACC-NOT,CHECK-HOST
+// RUN: env SYCL_DEVICE_FILTER=cpu,host %t.out | FileCheck %s --check-prefixes=CHECK-CPU,CHECK-HOST
```
Comment (vladimirlaz) [inline SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: The check for the absence of a filtered-out device was removed. It looks like a regression in the test.

**Review by vladimirlaz — CHANGES_REQUESTED** [review]

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp (hunk 2 above)
Comment (bso-intel) [reply SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: I did refactoring of get_devices(), and it returns CPU devices after GPU devices.

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp (hunk 3 above)
Comment (bso-intel) [reply SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: I splitted CHECK directive per each RUN command.

File: SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp (hunk 1 above)
Comment (bso-intel) [reply SYCL/FilterSelector/filter_list_cpu_gpu_acc.cpp]: added lots of CHECKS for different combinations now.

**Review by vladimirlaz — DISMISSED** [review]

**Review by smaslov-intel — DISMISSED** [review]

**Review by smaslov-intel — DISMISSED** [review]

**Review by vladimirlaz — DISMISSED** [review]

**Comment by bso-intel** [comment]
@vladimirlaz ,
Could you merge this PR as soon as possible since https://github.com/intel/llvm/pull/4247 is already merged?

**Pull request merged**